### ISSUE TITLE
Add Palemoon to Webbrowsers

### ISSUE THREAD

**Issue by test3333** [nylira/prism-break#1385]
Palemoon is a mature Firefox fork, that has gained its own traction.
Their development is poweruser and privacy focused and against adding things like drm, hello and pocket.
They also dont use the australis design.
It has grown enough to not be just a firefox mod anymore.
Because of addon incompatibilities they host their own addon site for cases where you cant use AMO, linked to on the main site.
Just in case some mozilla people come here spreading FUD like everywhere I saw Palemoon mentioned.
I was never a Palemoon advocate because I disagree with some of their practices but it is currently way more usable than Firefox for me. Lets not even speak of mozillas horrible practices in the last 2 years.
Palemoon uses sane defaults.

Main site: palemoon.org
Release Notes and History: http://www.palemoon.org/releasenotes.shtml
Linux Builds Page: http://linux.palemoon.org/
Sourcecode: https://github.com/MoonchildProductions/Pale-Moon

**Comment by alerque** [comment]
This suggestion is bothersome from a purely rhetorical perspective. Why worry about somebody else's FUD? Surely this isn't Palemoon's best foot forward. Can we just cut the nonsense? E.g. no more of this please:

> Lets not even speak of…

By saying that you _are_ speaking about that and, in fact, that is possibly the most relevant thing that could be spoken of when discussing the need for alternate web browser suggestions on Prism Break.

Is I have [said before](https://github.com/nylira/prism-break/issues/976#issuecomment-57429264), “PRISM-Break is not about listing open source alternatives to proprietary software.” Even less so is it about listing alternatives to alternatives for people who have different UI preferences.

So back to your suggestion:

> I was never a Palemoon advocate because I disagree with some of [Mozilla's] practices but it is currently way more usable than Firefox for me.

This interesting, but I'm not sure how relevant it is to the "PRISM issue". I understand people have different tastes in software, but being "less usable" is not generally an issue leveled against Firefox nor is an issue PRISM-Break is principally concerned with. Of course as much as possible we would like to recommend software that is more usable over software that is less usable, but the primary issue privacy and on-line security: escaping the PRISM drag net baked into many proprietary systems.

So I checked Palemoon's site so see what they are advocating and the most relevant bits seem to be interface related:

> This browser, even though fairly close to its sibling Mozilla Firefox in the way it works, offers a different set of features. For example, if you need specialized accessibility features or integration of Windows' parental controls, then Pale Moon may not be a good browser choice for you.

> With the current generation of Mozilla-based browsers, there are other differences between Pale Moon and Firefox that are more obvious and not just "under the hood": contrary to what Mozilla has done with their redesign of the user interface, Pale Moon will continue to provide a familiar set of controls and visual feedback similar to previous versions, including grouped navigation buttons of a decent size, a bookmarks toolbar that is enabled by default, tabs next to page content by default (easily switchable) and not in the least a functional status bar and more freedom in customization, to name a few things.

This lines up with what you were saying about usability. Perhaps for some users and use cases its behavior is more expected or better integrated with their other tools. Personally integration with Windows parental control is not something that lends itself to a PRISM break recommendation. Escaping the move to the Australis based UI is likewise completely tangential to what we're into here.

The Firefox UI is not my favorite either, but it's _not the problem we're here to address_. If there is a good case to be made for the browser from an Internet privacy and security angle I'd like to see _that case_ presented on its own merits without trying to make one project's UI design out to be the enemy.

That being said apart from the removal of Hello and some third party integrations, I'm not seeing much to go on. Am I missing a section of Palemoon's site that deals with this sort of issues? Do they have documented practices in regard to data privacy and inclusion of potentially leaky components? Are there any third party code audits or reviews by technical sources that could speak to how well those policies have been implemented?

**Comment by alerque** [comment]
Also it's probably worth cross referencing some previous discussions on browsers:

* Previous suggestion of Palemoon: #1220

   >  Pale Moon (and its Thunderbird fork FossaMail to some extend) looks like an extremely opinionated fork of Firefox. There is nothing wrong with that, of course, it is one of the great things about open-source software. They loved the old interface and didn’t change it and instead focussed on cutting out unneeded cruft to speed up the browser. But I see no mention of them working to enhance security. Your link says they will not implement DRM and that may be good from a philosophical standpoint but doesn’t change security one bit.

* Related to Firefox proper: #1184, #1307, #1355
* Related to other Firefox forks: #522, 
* Related to Chrome forks: #169, #799, #1175, #1309, #1311
* Other browser suggestions: #88, #524, #882, #1084

**Comment by mastercoms** [comment]
Doesn't Pale Moon have security issues though?

**Comment by alerque** [comment]
@mastercoms All software has security issues. You seem to be obtusely hinting at something more specific though. Could you actually be more specific and maybe provide some references? Whether software is a step forward or backward in relation to protecting users from digital surveillance is _the question_ that all new suggestions here are subjected to. Adding a comment to that effect is less useful than actually providing data would be.

**Comment by espress0** [comment]
Pale moon is great. No stupid pocket(closed-source), No Telemetary, No crash report, No Mozilla spyware.
+1 to adding this software to prism-break. Firefox is not an open source in these days.

https://github.com/MoonchildProductions/Pale-Moon/issues/179

**Comment by alerque** [comment]
@espress0 You're not helping your case. Just adding a +1 comment with a generic line calling things you don't like "stupid" comes across as FUD and doesn't further the discussion. If you really think Palemood should make the cut we need some detailed analysis of _why_ and hard evidence to answer the questions that have been raised about it. There are some detailed comments above and on linked issues that raise questions about the suitability of this recommendation. Unless you actually **address those concerns specifically** this isn't going to progress. And this isn't about how "bad" Mozilla is — you don't have to convince anybody that it does have some issues — what you need to do is show how Palemoon _is_ something to recommend. Not having pocked baked in is great, but that doesn't answer the questions about what it does or doesn't bring to the table that makes it suitable for general recommendation.

**Comment by wolfbeast** [comment]
You're free to audit the Pale Moon source code at any time that it is convenient for you to know exactly what it brings to the table. I (as lead dev and project owner) don't particularly see the need to invest time and effort into convincing you of its suitability for a recommendation, although I think our security would be considerable enough to warrant such a thing.

If there are any specific questions about pale Moon's security, please feel free to ask them.

**Comment by LimboSlam** [comment]
@alerque: Pale Moon brings forth some good security/privacy by removing all the extra bloat that  Firefox recently has integrated into their browser. Such as tons of telemetry code (a lot of this wasn't even included because they don't have the need to track us, which is why finding documented practices for this hard to locate) introduced by Pocket, Hello, IM, Share, Fingerprinting prefs (to note, canvas.poisondata is a security feature only found in Pale Moon; more recently the XSS Filter) and whole lot of WebRTC related code, not to mention quick reflexes to patching vulnerabilities. Just look through their release notes and I promise you'll find most of what @test3333 mention and I.  

Now I'm not going to add any references to this as I feel @test3333 needs to be the one to properly represent this, also @alerque if you do a bit more research (I'm a straight forward guy, so don't take this offensively) into this wonderful open source project, you'll find the information about what your asking for and lots of good reviews about Pale Moon. 

**Hint:** look at their source code, technical details page and around our forums  ;)

**Comment by ragues** [comment]
Further to the comments above, here are some posts highlight the Pale Moon developer's philosophy & practices:
Data collection: https://forum.palemoon.org/viewtopic.php?t=7788#p50661
Telemetry code removal: https://forum.palemoon.org/viewtopic.php?t=8894&p=59226#p63011
Geolocation: https://forum.palemoon.org/viewtopic.php?f=24&t=3658#p21255
Third party software: https://forum.palemoon.org/viewtopic.php?f=24&t=2837
Phoning home: https://forum.palemoon.org/viewtopic.php?f=24&t=631
DRM: https://forum.palemoon.org/viewtopic.php?f=4&t=4539#p27734
Rumor control: https://forum.palemoon.org/viewtopic.php?t=7818

**Comment by ragues** [comment]
... and XSS filter: https://forum.palemoon.org/viewtopic.php?t=10817#p75663

**Comment by LimboSlam** [comment]
I guess I should add this too as well.

**Why secure connections fail in recent versions of Pale Moon:** https://forum.palemoon.org/viewtopic.php?f=24&t=6262#p40401.

**Comment by geplus** [comment]
Pale Moon is really privacy oriented, higly recomended, I use it as my uniqe browser.
https://www.palemoon.org/
https://www.palemoon.org/licensing.shtml

Also, I think Pale Moon's redistribution license greatly protects the users' privacy, more than any other browser:
https://www.palemoon.org/redist.shtml

**Comment by Atavic** [comment]
I see JonDo, Firefox & Tor listed. Palemoon could sit well between Firefox and the two anonymity-aware options.

**Comment by yegortimoshenko** [comment]
Pale Moon censors their add-ons store: https://github.com/nylira/prism-break/issues/1802

Pale Moon's website has Google ads and tracking: https://github.com/nylira/prism-break/issues/1801#issuecomment-369763450

This is not enough to disqualify it from PRISM Break, although it is questionable whether Pale Moon's goals align well with this project.

**Comment by wolfbeast** [comment]
For the record:

1. We do not censor our add-ons store, but we DO by default block harmful extensions (either harmful to the browser user or harmful to innocent third parties). This behavior is easily adjusted from within the browser UI itself.
2. As far as ad providers go, Google is one of the least intrusive ad networks available with proper vetting of ad content (avoiding practically all malvertising) -- that's an objective fact. We'll be more than happy to remove the ad if our income loss from it is somehow compensated. "We're willing, but we need the income right now". If you decide to block it, we will show you a dismissible message once every 30 days when you visit the home page and not limit your visit otherwise.

**Comment by Hillside502** [comment]
We're not talking "malvertising", we're talking "tracking".

**Comment by Atavic** [comment]
>Pale Moon's website has Google ads and tracking

It's *their own* site and the choices taken are :+1: **IMHO**

Pale Moon censored a single and very debated addon (AdNauseam).

The browser totally aligns with this repo, we're calling out personal choices here.

**Comment by Hillside502** [comment]
Palemoon website has tracker(s).  Period.

**Comment by yegortimoshenko** [comment]
> [...] but we DO by default block harmful extensions (either harmful to the browser user or harmful to innocent third parties). 

This is by definition censorship. Harm in this case is subjective: it can also be argued that uBlock Origin is harmful to advertising industry. Blocking add-ons that are harmful is comparable to creating an operating system that limits what programs it can or can not run based on moral judgements of its developers.

> As far as ad providers go, Google is one of the least intrusive ad networks available with proper vetting of ad content (avoiding practically all malvertising) -- that's an objective fact. 

People here generally see a lot of problem with the fact that too much internet activity is leaked to Google. And activity from Google ads is unquestionably used to build a profile about the user.

I would additionally disagree that Google is one of the least intrusive ad networks. I think that would set too low of a standard: the only more intrusive ads I've ever encountered are those on porn and warez sites.

Native advertisement (no third parties) is obviously the best approach in this regard. Other than that, I think Project Wonderful is much better, in transparency and in JS pushed to client. There also was [The Deck](http://decknetwork.net/), perhaps someone can point me to other nice options.

Again, advertisement is not the problem, but reporting activity to Google, and using that data is.

**Comment by Atavic** [comment]
The site has trackers, Yes. But we list the browser here.
I mainly agree with points listed, but I use/like and will promote *the browser*.

**Comment by yegortimoshenko** [comment]
> We'll be more than happy to remove the ad if our income loss from it is somehow compensated.

I understand and sympathize with that. My personal advice would be to set up [Liberapay](https://liberapay.com/) ([demo](https://liberapay.com/Liberapay/donate)) or Patreon ([demo](https://www.patreon.com/cthulhoo)), both platforms are specifically for recurring donations (it's not possible to send a one-time donation).

One-time donations are problematic in that people might still receive value from you (i.e. Pale Moon updates), but they are used to having Pale Moon so they forget about donating to you. PayPal does have recurring donations, but it's opt-in, and implemented in an afterthought manner, there are no funding goals/tiers, etc.

**Comment by Atavic** [comment]
:+1: for Patreon

**Comment by hasufell** [comment]
> Google is one of the least intrusive ad networks

Google is the biggest ad tracking system out there.

**Comment by LimboSlam** [comment]
Um... you guys.. GitHub has tracker too (google-analytics.com) and I don't see anyone complaining. In fact, every website on the Internet have advertisements for income or trackers to improve UX. 

Just a thought....

**Comment by yegortimoshenko** [comment]
See https://github.com/nylira/prism-break/issues/767.

**Comment by wolfbeast** [comment]
Regardless of your dislike of Google, Pale Moon AS A BROWSER does not include any Google ads or trackers. You're excluding a piece of software because its homepage (ass opposed to the actual software) runs an ad? I think you should re-think where your priorities lie in that case. The question here is not whether to "add the Pale Moon website", the question is "whether to add the Pale Moon browser".

In fact, if you have Firefox on your list be aware that Firefox DOES include Google tracking _inside_ the browser in various spots, not all of which are selectable or opt-in (geolocation, safebrowsing, pings sent from the add-ons including full enumerated lists of everything installed, etc. etc.)

>>    [...] but we DO by default block harmful extensions (either harmful to the browser user or harmful to innocent third parties). [...]

>This is by definition censorship. Harm in this case is subjective: it can also be argued that uBlock Origin is harmful to advertising industry, that you're "stealing" sites when you load them without ads, similar to "piracy".

If that is your stance, then you need to remove all browsers that employ any sort of blocklist for harmful extensions, since by your definition, any blocklist equals censorship. You should either apply your stance evenly, or not at all, because otherwise you're just being biased and your whole premise behind this project is going to be false.

**Comment by yegortimoshenko** [comment]
> In fact, if you have Firefox on your list be aware that Firefox DOES include Google tracking inside the browser in various spots, not all of which are selectable or opt-in (geolocation, safebrowsing, pings sent from the add-ons including full enumerated lists of everything installed, etc. etc.)

Google is one of two "safe browsing" providers, and it also doesn't send any metadata about the user beyond the fact that someone on this particular IP address is using this particular browser.

This is very different from including Google ads on download page which gathers data on what sites user is interested in, how they interact with the content, visit duration, and can include canvas fingerprint, battery API timing, locale, installed fonts, timezone... In comparison, with safe browsing DB access, the only metadata that is ever leaked is User-Agent. The same applies to geolocation API.

However, "Get add-ons" page does still, to my knowledge, use Google Analytics: https://github.com/mozilla/addons-frontend/issues/2785. Mozilla has signed a contract with Google that prohibits them from using this data in any way, but this does not make it OK. This is where Pale Moon is better than Firefox from privacy perspective.

> If that is your stance, then you need to remove all browsers that employ any sort of blocklist for harmful extensions, since by your definition, any blocklist equals censorship. You should either apply your stance evenly, or not at all, because otherwise you're just being biased and your whole premise behind this project is going to be false.

I should make it clear: this is not my personal project. I'm not in position to remove something based only on my opinion.

I don't think we list any browsers that block add-ons based on their moral value. It's OK to maintain black list of extensions that trick user into unknowingly installing malware (e.g. credit card sniffer), that would be malware protection. AdNauseam does not hide any malicious functionality directed towards user, and is actually very transparent in what it does, here's their Firefox Add-ons page: https://addons.mozilla.org/en-US/firefox/addon/adnauseam/

When program decides for the user what they want or don't want to install, it is censorship: I think it is definitional and not subjective, from [Wiktionary](https://en.wiktionary.org/wiki/censor#English):

> From Latin *cēnsor* (“magistrate, critic”), from *censere* (“to tax, assess, value, judge, consider, etc.”).
> 1. To review in order to remove objectionable content from correspondence or public media, either by legal criteria or with discretionary powers. 
> 2. To remove objectionable content.

As I said before, I do not believe this to be a flaw that would disqualify Pale Moon, but in my opinion, add-on censorship is dangerous and is something to be considerate about. It is nice that it can be disabled but I don't understand the value in that from user perspective.

**Comment by yegortimoshenko** [comment]
Unfortunately, Pale Moon has even more draconian redistribution agreement than Firefox. See: https://www.palemoon.org/redist.shtml

Please compare it side-by-side with [Firefox Trademark Policy](https://www.mozilla.org/en-US/foundation/trademarks/policy/) (start from Software Distributions subtitle).

Redistribution of self-built Pale Moon under the same name and logo is not allowed by this agreement, with the following exception:

> An exception applies to point 5 of this license for inclusion of the officially branded binaries in freely available and fully Open Source operating systems, including but not limited to non-commercial variants of Linux, variants of BSD and ReactOS. This exception only applies to unaltered versions of the Pale Moon binaries or officially branded variants specifically built for the target operating system from source that have not been materially changed (including build configuration not necessary for the resulting binary's proper operation and brand-specific configurations like e.g. home page, default search engine). If any of the essential settings of the browser are altered beyond what is strictly needed for providing a working and stable build on the target operating system, the exception in this point does not apply and the license defaults to point 10, instead. 

Meaning, even applying CVE fixes that were commited but have not been released is prohibited, any kind of patches except those that fix the build. Firefox has a similar policy, but they allow any changes that are not "significant functional changes", which might be vague but certainly much more lax than Pale Moon.

>  If an officially-built binary (i.e. built by our team) is available for the target operating system (e.g. existing Linux binaries), then those binaries will always, in principle, take precedence over a third-party build to ensure necessary QA and compatibility. Building and distributing a binary with official branding is in that situation only in principle allowed if an actual variant build is required for the target distribution's compatibility (e.g. kernel or library requirements) or operation, and otherwise not impacting the material content of the browser package as a whole. QA of and support for the resulting variant binaries will in that case fall on the maintainer who will have to take full responsibility for the variant binary.

Meaning, Debian can't package Pale Moon to ensure a fully reproducible build, user has to rely on an third-party Debian repo endorsed by @wolfbeast: https://linux.palemoon.org/

Also, that means that any distribution that risks adding Pale Moon to main repo will be breaking this agreement if @wolfbeast starts distributing binaries for that distribution.

Also, distributing *any* builds for Windows or macOS other that those made by Pale Moon team is not allowed. This is sad because there are package managers that build everything from source for proprietary systems (Gentoo Prefix, Nix, partially homebrew-cask) and they can't include Pale Moon under its proper name. Package management is useful on every platform and guarantees that executable corresponds to source code.

---

This is a major problem, that means Pale Moon on Linux, FreeBSD and OpenBSD cannot be fully supported by those distributions, and can't receive the same protections (in some cases, reproducible builds, Debian applies heavy patches to almost everything they distribute, etc.) Users on those platforms should also trust that Pale Moon's build servers are not compromised, unlike with Linux/FreeBSD/OpenBSD.

**Comment by yegortimoshenko** [comment]
Now the fun part. From [Pale Moon's branding page](https://www.palemoon.org/branding.shtml):

> The Pale Moon branding, logo, and project/product names are the Intellectual Property of M.C. Straver, the owner and lead developer of the project. Pale Moon and the phrase "Your browser, Your way" are Trademarks ™ for the web browser and the usage rights of the name and logo are reserved by the owner.

Could I ask, do you actually hold a trademark on Pale Moon? [I couldn't find one in US](http://tmsearch.uspto.gov). [I couldn't find one in Sweden](https://was.prv.se/VarumarkesDb/searchMain.jsp), where I think you reside. Also, trademarks [are territorial](https://www.stopfakes.gov/article?id=Does-a-U-S-Trademark-Registration-Protect-a-Trademark-in-a-Foreign-Country) and should be filed for each individual country.

(By the way, Pale Moon [was a valid trademark in US from October 2008 up until May 2015](https://tsdr.uspto.gov/#caseNumber=77173561&caseType=SERIAL_NO&searchType=statusSearch). [Wikipedia says](https://en.wikipedia.org/wiki/Pale_Moon_(web_browser)) Pale Moon the browser has appeared in 2009! :rabbit2:)

It seems to me that branding is not protected under trademark law. I would love to be proven wrong, please do. Some Wikipedia editors [are also questioning that there are trademarks in place](https://en.wikipedia.org/wiki/Talk:Pale_Moon_(web_browser)#No_citation_to_registered_trademarks).

Names cannot be protected under copyright law, and logo [seems to be available under CC-BY-SA](https://commons.wikimedia.org/wiki/File:Pale_moon_logo_256.png).

---

I'm saying all of that because I don't like that Pale Moon witch hunts builds of its browser that include locales other than en-US (titled "Illegal redistributions"): https://forum.palemoon.org/viewtopic.php?f=1&t=657

"Illegal" is not a word to be said lightly. Could you please argue why is that illegal? You do not provide non-English builds, but most of population of this planet does not speak English. Some people seem to be kind enough to build it for those that will be more comfortable with their native locale.

Also, from your redistribution agreement:

> This has been required because of, among other things, the increasing number of rogue/altered copies and people taking advantage of the free availability of the browser to monetize upon, which is against Pale Moon's principles of free software.

I think you misunderstand what free software is. Free software is free as in freedom, not as in beer. Monetizing upon software is just fine.

**Comment by yegortimoshenko** [comment]
I am against inclusion of Pale Moon for the following reasons:

* Tiny developer base in comparison with Firefox
* Google ads on download page + redistribution policy make it hard to download the browser without getting tracked; also a very dangerous business model for a browser (it pays off to have as many visits as possible, i.e. for example set Pale Moon's website as home page)
* add-on censorship based on moral judgement
* proprietary binaries with only small subset of redistribution rights, draconian redistribution agreement that prohibits from proper inclusion in software distributions

These I think most people here would agree with. Perhaps somewhat more controversial, I also really dislike when people baselessly call something illegal, this appears to be scare tactics to me.

**Comment by Atavic** [comment]
To me, only the last point stands. A small number of devs is better than Mozilla because the latter are included in a pyramidal scheme where most of the user-base issues are postponed and then ignored. The site is their own, while I'm looking at the browser that's available on many different D/L sites.

For the rejected addons, one can work around the censorship, see:
https://github.com/dhowe/AdNauseam/issues/1121

I hate legalese, so the last point - eventually - stands.

**Comment by Atavic** [comment]
To me, only the last point stands. A small number of devs is better than Mozilla because the latter devs are included in a pyramidal scheme where most of the user-base issues are postponed and then ignored. The site is their own, while I'm looking at the browser that's available on many different D/L sites.

For the rejected addons, one can work around the censorship, see:
https://github.com/dhowe/AdNauseam/issues/1121

I hate legalese, so the last point - eventually - stands.

**Comment by yegortimoshenko** [comment]
> To me, only the last point stands. A small number of devs is better than Mozilla because the latter devs are included in a pyramidal scheme where most of the user-base issues are postponed and then ignored. The site is their own, while I'm looking at the browser that's available on many different D/L sites.

Do you mean this is a hard fork of Gecko? With one full-time developer, Pale Moon is bound to become obsolete within several years. If Pale Moon still pulls patches from Gecko, or shares dependencies developed by Mozilla, then you're still indirectly using work of those 100+ developers (pessimistic estimate, Mozilla [had 1000+ employees worldwide in 2016](https://github.com/MoonchildProductions/Pale-Moon/commits/master)).

---

Unfortunately, Google ads do not end on the website. I've just installed Pale Moon and first run page has a banner: 

![screenshot_2018-03-09_23-04-52](https://user-images.githubusercontent.com/1643293/37234544-4c78bd3e-23f0-11e8-9ec6-2e893844f692.png)

And search page (that opens by default) has a banner:

![screenshot_2018-03-09_23-10-47](https://user-images.githubusercontent.com/1643293/37234546-535c07a0-23f0-11e8-8cd3-0c5020c8f7c7.png)

I am surprised. This is adware, incompatible with goals of this project.

**Comment by wolfbeast** [comment]
Still misinformed about the following:
- Single developer? Nope.
- Hard fork of gecko? yes, absolutely.
- Website != application. Website is also != just the index page.
- Start portal ads? The portal was a request BY OUR USERS to have a fully configurable start page. It's a home page. Change it if you want.
- Redistribution license allows any redistribution through any channel as long as unaltered, AND in fact requires that download sites do NOT gather personal information or slap on tracking/ad stubs. How you conclude the opposite is beyond me. Read it again.
- No add-on censorship and free user choice to allow add-ons that are harmful. No draconian add-on signing policy either like Mozilla, if you want to talk about draconian/censorship.

But, feel free to close this issue if all you intend to do is argue instead of actually considering inclusion. You are obviously not going to waver from your stance and I at this point would prefer to have Pale Moon **not** be part of such a single-minded, closed, biased "effort" that turns out to be no more than a group of people who want to impress their views on the world with no regard for the reality of software development and internet economy around providing an application both libre and gratis.

**Comment by yegortimoshenko** [comment]
> Website != application. Website is also != just the index page.

Would you argue that first run page is also a website and not part of the application? It also shows Google ads, like every other page in Pale Moon, and like every page bookmarked by default except forums.

And defaults are very important. Pale Moon defaults to adware and selling user data to Google. This is worse than any other browser ever, not just those that are listed on this website.  

> Redistribution license allows any redistribution through any channel as long as unaltered, AND in fact requires that download sites do NOT gather personal information or slap on tracking/ad stubs. How you conclude the opposite is beyond me. Read it again.

Read 8a. Quote, "Building and distributing a binary with official branding is in that situation only in principle allowed if an actual variant build is required for the target distribution's compatibility".

This means it's not really possible to distribute Pale Moon in a distribution without you potentially forcing it out at some point (or forcing it to be renamed to New Moon).

You also use this agreement to enforce banners on default pages (you even list every place that incidentally includes banners as an example of what should not be altered):

> This exception only applies to unaltered versions of the Pale Moon binaries or officially branded variants specifically built for the target operating system from source that have not been materially changed (including build configuration not necessary for the resulting binary's proper operation and brand-specific configurations like e.g. home page, default search engine)

And then you spread FUD on forums that builds made by someone other than you "have a large risk of having malicious payloads".

> No add-on censorship and free user choice to allow add-ons that are harmful.

This feature was designed to block malware. You use it for censorship. Error page doesn't even say how to turn it off. You've put the switch into Security section and called it "Add-on security level", even though it is not at all related to security, creating impression that toggling the switch to allow AdNauseam will cause browser to be less secure.

> No draconian add-on signing policy either like Mozilla, if you want to talk about draconian/censorship.

So, no AMO? Then it should be trivial to distribute malware through your add-ons store.

> Single developer? Nope.

Single *full-time* developer. It seems that you are doing 80% of the work: https://github.com/MoonchildProductions/Pale-Moon/commits/master

> You are obviously not going to waver from your stance and I at this point would prefer to have Pale Moon not be part of such a single-minded, closed, biased "effort" that turns out to be no more than a group of people who want to impress their views on the world with no regard for the reality of software development and internet economy around providing an application both libre and gratis.

You are selling your users, I don't want PRISM Break visitors to become your users for you to sell.

Your redistribution agreement is so strict that software is effectively proprietary, read about djb and his licensing policy and how distributions stopped packaging his software: [[1]](https://web.archive.org/web/20070824065650/http://cr.yp.to/distributors.html) [[2]](https://developers.slashdot.org/comments.pl?sid=351235&cid=21252141)

Also, given that you haven't commented on trademarks, I should assume that you lie on your website and this branding trademark policy holds no legal weight.

**Comment by Atavic** [comment]
>Would you argue that first run page is also a website and not part of the application?

I always modify user.js to prevent such connections. Changing th Home Page is a few clicks from the GUI.

>And defaults are very important.

Indeed, PM has better defaults than FF.

>Then it should be trivial to distribute malware through your add-ons store, or outside of it.

PM addons are indexed after a request to the Add-ons Team, via the official PM forum. So, there's humans controlling the addons before they are indexed.

**Comment by Atavic** [comment]
>Would you argue that first run page is also a website and not part of the application?

Changing the Home Page is a few clicks from the GUI.
I always modify user.js to prevent such connections. 

>And defaults are very important.

Indeed, PM has better defaults than FF.

>Then it should be trivial to distribute malware through your add-ons store, or outside of it.

PM addons are indexed after a request to the Add-ons Team, via the official PM forum. So, there are humans controlling the addons before they are indexed.

**Comment by mattatobin** [comment]
According to https://www.gnu.org/philosophy/free-sw.html

> Rules about how to package a modified version are acceptable, if they don't substantively limit your freedom to release modified versions, or your freedom to make and use modified versions privately. Thus, it is acceptable for the license to require that you change the name of the modified version, remove a logo, or identify your modifications as yours. As long as these requirements are not so burdensome that they effectively hamper you from releasing your changes, they are acceptable; you're already making other changes to the program, so you won't have trouble making a few more. 

Thus:

> Your redistribution agreement is so strict that software is effectively proprietary

Is the big lie here.

Also see: https://opensource.org/licenses/mpl-2.0

> 2.3. Limitations on Grant Scope
> [...]
> This License does not grant any rights in the trademarks, service marks, or logos of any Contributor (except as may be necessary to comply with the notice requirements in Section 3.4).

and

> 3.2. Distribution of Executable Form
> [...]
> You may distribute such Executable Form under the terms of this License, or sublicense it under different terms, provided that the license for the Executable Form does not attempt to limit or alter the recipients’ rights in the Source Code Form under this License.

Thus: Rights to trademarked branding (which do not have to be registered to be valid) are ONLY extended by the Pale Moon Redistribution license. Not by the MPL for which the Redistribution license its self is allowed by the MPL and in any event recognized by GNU's guidelines of Free and Open Source software as being not only valid but also reasonable.

So wanna try again?

**Comment by yegortimoshenko** [comment]
@Atavic: you can either close your eyes on Pale Moon engaging in surveillance capitalism, something we're fighting here, or own the fact that Pale Moon has built its business model on surveillance. I'm sure you'd treat it quite differently if Firefox went ahead and placed Google ads on first run page.

> So, there are humans controlling the addons before they are indexed.

If Pale Moon allows untrusted add-ons, this is a major downgrade in security. Pale Moon should sign add-ons after review to compete with Firefox in this regard.

**Comment by yegortimoshenko** [comment]
@mattatobin: I know that MPL allows licensing binaries under any license, if there is no trademark, this license can be circumvented just by rebuilding said binary from source.

> Is the big lie here.

Is it, though? I don't know a single major distribution that includes Pale Moon under its proper name in the main repo. There is also a precedent to believe otherwise: read about Debian and Iceweasel. Sure, source code is not proprietary, but the brand is (or you pretend it to be, rather). It's not possible to compete with you without building audience for another brand (say, New Moon).

> Thus: Rights to trademarked branding (which do not have to be registered to be valid) 

It doesn't work this way :-) Your "trademark" is unenforceable. You also clump together redistribution agreement (e.g. license for proprietary binaries, as allowed by MPL) and "trademark" policy (e.g. when it can and cannot be used without prior authorization).

**Comment by mattatobin** [comment]
Ok.. We have TWO Adblockers specifically created for Pale Moon both based on Adblock Plus..

Adblock Latitude is maintained by the project.. ABPrime is maintained by my SELF who does a significant amount of work on the Project..

Our request on the Pale Moon sites to allow ads on them states:

> We love ad blockers as much as you, but we depend on ad revenue to fund various sites and services.
> We use responsible ad services to keep your visit to our websites a safe and uninterrupted one.
> To ensure our continued operation, please disable your ad blocker for this site or support us another way.

It is totally up to the user to unblock ads or not or to help fund the project in another way.. So what is the problem.. We have two adblockers that DO NOT have defaults that unblocks Pale Moon's sites or anything.. We like adblocking so much we have TWO of them lol..

So what is the problem?

**Comment by mattatobin** [comment]
The redistribution agreement extends permission to distribute unmodified binaries and a few other provisions which extend permission.

Also, you are saying common law trademarks aren't enforceable? Since when?

**Comment by yegortimoshenko** [comment]
> So what is the problem?

The problem is that you default to ads (opt-out rather than opt-in), it's not possible to download and even open the browser (say, to download the adblocker) without getting tracked, you censor addons that are particularly harmful for participants of advertisement industry (AdNauseam), and your business model is built on serving as many ads as possible.

**Comment by mattatobin** [comment]
Well it is kinda harmful.. It creates ad fraud by default.. But one can adjust the settings to allow installation of that specific extension.

**Comment by yegortimoshenko** [comment]
> The redistribution agreement extends permission to distribute unmodified binaries and a few other provisions which extend permission.

I know, but this is very tiny subset of natural rights on redistribution (only non-commercial, not bundled, not altered in any way, not even unarchived, et cetera). This is not reasonable at all, Firefox's trademark policy is problematic enough, and yours is much, much stricter.

> Also, you are saying common law trademarks aren't enforceable? Since when?

It depends from country to country, Russia (my country of residence) for example doesn't recognize common law trademarks. Also, I think there has been no legal precedent in internationally enforcing common law trademarks.

Common law trademarks in US receive a very small subset of protection in comparison with registered trademarks, mostly only protection against misinterpretation of source of origin only within specific state, which is not what you're trying to enforce in your trademark policy.

**Comment by yegortimoshenko** [comment]
> Well it is kinda harmful.. It creates ad fraud by default.. But one can adjust the settings to allow installation of that specific extension.

That's one way to look at it. Another way is that it allows user to effectively destroy their advertisement profile by filling it with noise and thus making it useless. It also fights with internet tracking, forcing projects to find a more ethical way to make money. This should be decided by user, not by browser vendor who's incidentally making money off ads.

**Comment by yegortimoshenko** [comment]
> Doesn't matter anyway, since we are such capitalist surveillance pigs and all.. I dunno why this is a conversation you are continuing to have..

I'm just responding to you, and waiting for other contributors to jump in and make the final decision. I can't unilaterally close this issue. 

> Hell, if you DO want to use our work just don't use our logos and name and it will be fine.. 

Are you asking me or the project? This project has no policy on IP other than that free software is good. People are all over the spectrum here and have different stances on the issue, use very different licensing for their own works, etc.

My own opinion is that intellectual property is illegitimate and you have no right to make words, or logos your own. In context of your project, you pretend to use IP laws to enforce that all copies of your software contain banners and tracking. You also hunt down people who add features to your product under the same name, like locales. This is antisocial and in my perspective not OK at all.

**Comment by Atavic** [comment]
>I don't know a single major distribution that includes Pale Moon under its proper name in the main repo.

Slackware main repo has it.

**Comment by mattatobin** [comment]
Well in the real world, copyright and intellectual property is a thing.. We mostly treat it as an identity.. Our collective name.. How would you like me, to just go out and call my self `yegortimoshenko` or whatever your real name is and misrepresent everything you believe in or modify work you have done and just claim it as my own?

When people come and bother you for something I did in your name would that be a problem for you?

**Comment by mattatobin** [comment]
> I don't know a single major distribution that includes Pale Moon under its proper name in the main repo.

Manjaro
PCLinuxOS
Puppy Linux
MEPIS/MX-15
Gentoo Overlays
Slackbuilds

**Comment by yegortimoshenko** [comment]
None of the distributions are major, or even significant. The fact that they can't patch anything they may find questionable (like what Debian does) is definitive, because major distributions have QA policies that require being able to patch source code or build it with custom options to enforce security mitigations.

> Well in the real world, copyright and intellectual property is a thing.. 

That's rather strange thing to say when you've forked a copyleft project (and copyleft is quite anti-IP, it forces on derivatives to not apply copyright law, basically).

> How would you like me, to just go out and call my self yegortimoshenko [...]

Well, sure, you can. Fake accounts are not illegal anywhere in the world :-) Most famous people have this problem, it would be ridiculous of them to sue owners of hundreds of fake accounts on, say, Twitter. There are systems to identify real people from impostors, like Web of Trust (which can be enforced by protocol like in GPG or be implicit), or putting your account name on trusted place like your website, etc.

> [...] modify work you have done and just claim it as my own.

You can skip attribution, but misrepresenting origin is usually an offense that is defined outside of intellectual property laws (usually known as "plagiarism"), it even applies to public domain works.

Anyway, I don't see your site arguing for that on moral ground, you prefer to scare people into believing they might get sued. I think I've provided enough proof that your trademark policy does not hold any legal weight.

Also, if you don't want to continue arguing with me, just don't reply, please. If I lock this discussion, that will cause everyone without contributor access to not be able to reply, which I don't want to do.

**Comment by mattatobin** [comment]
Just to let you guys know how much we could do and be within our rights as well as compliant with the MPL.

We COULD simply remove our public repos and require anyone who wants the source code to have to us for a physical DVD or thumbdrive with the base Mozilla source code plus every commit we have applied as patch files.

According to the MPL Section 3.1 point a:

> such Covered Software must also be made available in Source Code Form, as described in Section 3.1, and You must inform recipients of the Executable Form how they can obtain a copy of such Source Code Form by reasonable means in a timely manner, at a charge no more than the cost of distribution to the recipient;

Which means you would have to pay for the physical medium of our choice and postage. Of course we don't do that.. But we could.

**Comment by Atavic** [comment]
I struggle following licenses issues, but...

[JonDoFox](https://prism-break.org/en/projects/jondofox/) Licence:

>Copyright (c) The JAP-Team, JonDos GmbH

So, should I open an issue about its license?

**Comment by yegortimoshenko** [comment]
> So, should I open an issue about its license?

Please keep this issue on topic :-) No need to ask to open a new one, just open, never a problem.

**Comment by mattatobin** [comment]
@Atavic Looks like a BSD style license so as long as they keep copyrights and don't use the project name or contributors to endorse the project.. it should be fine.

**Comment by yegortimoshenko** [comment]
Another example of Pale Moon bullying developers: https://github.com/jasperla/openbsd-wip/issues/86

**Comment by hasufell** [comment]
> Another example of Pale Moon bullying developers:

It's not that simple. I've been a packager for gentoo and exherbo for 6 years. What happened over at that bug report you linked **is** a pretty huge oversight by the packager. It's his job to know how to package and redistribute something. Don't be surprised that something unpleasant happens if you mess that up. That's why e.g. Gentoo has a [license team](https://wiki.gentoo.org/wiki/Project:Licenses) to help with such questions/situations.

Although, I agree that the aggressiveness by the Pale Moon developers is a bit disproportional. In a way that makes me think I'd better avoid that project, even for contributing patches.

**Comment by mattatobin** [comment]
I have already admitted that I should have switched a few words around.. Mainly, "You will" to "Will you" but the aggression back is what escalated this beyond reason.

Mistakes and missteps in tone were made and obviously, even with my phrasing here changes have happened. I'd like to thing that is apparent. Though, it may not be and I need to try harder.

**Comment by hasufell** [comment]
> is the person in charge of the Pale Moon overlay we consider to be official

That ebuild has not been reviewed by a gentoo dev (I easily spot various things from a quick look at it). This is the problem that arises when you do things on your own. So I consider it unofficial ;)

**Comment by yegortimoshenko** [comment]
I've been a Gentoo packager for a year and I am currently a NixOS developer.

Read the whole thread. They have been nicely approached by the packager two days before this ticket: https://forum.palemoon.org/viewtopic.php?t=18256

`openbsd-wip`, as the name implies, does not distribute packages, as such bullying is absolutely unwarranted. It is also unwarranted because Pale Moon is not a registered trademark and can't be enforced internationally as such.

**Comment by mattatobin** [comment]
Here is a question. Why are you out to destroy our project over this. Did you actually read that thread after The first post?

**Comment by mattatobin** [comment]
Additionally and as a matter of record.. The OP in that thread was not seen by my self nor Moonchild until after that incident took place.

All I knew at the time was someone said on IRC that they were using branded builds on BSD. I did a search and that was the first result. I checked the configuration and made that opening post.

**Comment by mattatobin** [comment]
Let us for a second assume common law trademarks are somehow unenforceable or invalid. Wouldn't that then fall back to personal copyright for the artwork?

**Comment by alerque** [comment]
@yegortimoshenko Please tone it down a bit. I appreciate that old issues are getting rummaged through and cleaned up (that's super valuable) but as I'm going through keeping up with notifications it's disconcerting to come across stuff like this in our issue tracker. Your tone is far too aggressive in this case and I feel does not represent PRISM-Break well as a project itself. Obviously your entitled to your opinion and input as a developer, but particularly while wielding a `collaborator` badge I don't think this kind of aggressive engagement with upstream projects is good for us.

I'm not even trying to advocate for Palemoon inclusion here, I'd tend to veto it myself, but I feel like that issue got lost in an tangential bickering debate that didn't need to happen we owe the other side of this a bit of an apology.

**Comment by Sa-Ja-Di** [comment]
@yegortimoshenko 

Speaking about "selling users":

One company has really experience in that.

Mozilla.

Doing that since 2013 - Betraying peoples trust (add-on/theme developer and power users) because they want to be so badly to be like Google/Chrome.

THAT is a serious betrayal and selling in big style.

**Comment by Hillside502** [comment]
For "rummaged through and cleaned up" read "methodically pruned and updated".

Previous to Yegor, PRISM Break appears to be run by jacks-of-all-trades (by their own admission), masters-of-none.  Notably, the founder has bottled out.  No use having a dog and barking yourself!

Three cheers for Yegor!

**Comment by yegortimoshenko** [comment]
No one seems to have any objections on the matter of subject, so closing, based on hostile trademark law policy that makes it impossible to integrate Pale Moon into a mainstream distribution by prohibiting modifications, and ad tracking inside the browser.

**Comment by hasufell** [comment]
> No one seems to have any objections on the matter of subject

What? Have you read any of the comments?

**Comment by yegortimoshenko** [comment]
If you have objections that would make it worthy of inclusion to PRISM Break, please tell.

**Comment by mattatobin** [comment]
Just for the record.. There is no Advertising in the software its self but there are ads on the websites the project uses.. We also find this kinda unacceptable.. Thus the project as launched a Patreon.. If we can reliably get about $400 a month we can completely remove ads from all the Pale Moon sites completely.. If we can get 700 dollars we can drop the sponsors as their time runs out.

https://www.patreon.com/MoonchildProductions

Otherwise, the ads have to stay on the websites.. Of course, one does not have to use any of the Pale Moon sites or start.me to use the software.

Anyway, this is a closed topic and I am gonna bow out now..

Peace!

**Comment by hasufell** [comment]
> If you have objections that would make Pale Moon worthy of inclusion to PRISM Break, please tell.

First, you start by reopening the discussion. Your bias is becoming a problem here. There was no clear consensus in this discussion anywhere and you just close it based on your personal opinion.

**Comment by yegortimoshenko** [comment]
Do you have any specific objections, though? As in, do you see Pale Moon on the list in its current form? I have waited three weeks for feedback from other members, @alerque ultimately agreed in that it should be vetoed.

**Comment by hasufell** [comment]
> Do you have any specific objections, though?

First, start by reopening the issue. Your behavior doesn't reflect how we reach consensus here.

**Comment by hasufell** [comment]
> I have waited three weeks for feedback from other members

This is a community.

**Comment by macandchief** [comment]
All I'd like to add is that default Firefox seems to be spyware, while Pale Moon is not.

**Comment by macandchief** [comment]
...sorry, a link is missing: https://spyware.neocities.org/articles/browsers.html

**Comment by alerque** [comment]
This is not the current issue tracker, discussion does not continue here.
